src/effects/Blur.cpp

```diff
@@ -74,136 +74,147 @@ std::shared_ptr<Frame> Blur::GetFrame(std::shared_ptr<Frame> frame, int64_t fram
-	for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) blur_blue[i] = blue[i];
-	for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) blur_alpha[i] = alpha[i];
+	// Initialize target struct pointers for boxBlur operations
+	channels *chans_a = &arrays_a;
+	channels *chans_b = &arrays_b;
+	channels *output = chans_a;
 
 	// Loop through each iteration
-	for (int iteration = 0; iteration < iteration_value; iteration++)
+	#pragma omp parallel for
+	for (int iteration = 0; iteration < iteration_value; ++iteration)
 	{
-		// HORIZONTAL BLUR (if any)
-		if (horizontal_radius_value > 0.0) {
-			// Init boxes for computing blur
-			int *bxs = initBoxes(sigma_value, horizontal_radius_value);
-
-			// Apply horizontal blur to target RGBA channels
-			boxBlurH(red, blur_red, frame_image->width(), frame_image->height(), horizontal_radius_value);
-			boxBlurH(green, blur_green, frame_image->width(), frame_image->height(), horizontal_radius_value);
-			boxBlurH(blue, blur_blue, frame_image->width(), frame_image->height(), horizontal_radius_value);
-			boxBlurH(alpha, blur_alpha, frame_image->width(), frame_image->height(), horizontal_radius_value);
-
-			// Remove boxes
-			delete[] bxs;
-
-			// Copy blur_<chan> back to <chan> for vertical blur or next iteration
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) red[i] = blur_red[i];
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) green[i] = blur_green[i];
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) blue[i] = blur_blue[i];
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) alpha[i] = blur_alpha[i];
-		}
-
-		// VERTICAL BLUR (if any)
-		if (vertical_radius_value > 0.0) {
-			// Init boxes for computing blur
-			int *bxs = initBoxes(sigma_value, vertical_radius_value);
-
-			// Apply vertical blur to target RGBA channels
-			boxBlurT(red, blur_red, frame_image->width(), frame_image->height(), vertical_radius_value);
-			boxBlurT(green, blur_green, frame_image->width(), frame_image->height(), vertical_radius_value);
-			boxBlurT(blue, blur_blue, frame_image->width(), frame_image->height(), vertical_radius_value);
-			boxBlurT(alpha, blur_alpha, frame_image->width(), frame_image->height(), vertical_radius_value);
-
-			// Remove boxes
-			delete[] bxs;
-
-			// Copy blur_<chan> back to <chan> for vertical blur or next iteration
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) red[i] = blur_red[i];
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) green[i] = blur_green[i];
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) blue[i] = blur_blue[i];
-			for (int i = 0; i < (frame_image->width() * frame_image->height()); i++) alpha[i] = blur_alpha[i];
-		}
+		#pragma omp parallel num_threads(4) shared (chans_a, chans_b, output)
```

Comment:
Hey, one question, if anyone knows — this isn't something I was able to find in the docs so far, though I'll keep looking.

If this is run on a system with _fewer_ than 4 cores, will it still execute with exactly four threads? The way I have the code written, I need that to happen, because the thread ID controls which channel that thread processes. There have to be exactly four, no more no less, or it won't work right.

Reply:
Never mind — completely re-worked how OMP parallel processing is applied, and eliminated that bottleneck.